app/controllers/api/console.php

```diff
@@ -111,3 +115,29 @@
 
         $response->chunk('', true);
     });
+
+App::get('v1/console/resources/:resourceId')
+    ->desc('Check resource ID availability')
+    ->groups(['api', 'projects'])
+    ->label('scope', 'projects.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN])
+    ->label('sdk.namespace', 'console')
+    ->label('sdk.method', 'checkResourceAvailability')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_NONE)
+    ->label('abuse-limit', 10)
+    ->label('abuse-key', 'userId:{userId}')
+    ->param('resourceId', '', new UID(), 'ID of the resource.')
+    ->param('type', '', new WhiteList(['rules']), 'Resource type.')
+    ->inject('response')
+    ->inject('dbForConsole')
+    ->action(function (string $resourceId, string $type, Response $response, Database $dbForConsole) {
```

Comment:
this will allow the console to check resources that belong to other projects and users as well. is this expected ?

Reply:
Yes, and that's why the endpoint doesn't return the document, just the status code to reflect whether the resource is present or not